src/models/config.rs

```diff
@@ -157,11 +169,47 @@ impl Repo {
                 target_theme.version = theme.version.clone();
                 target_theme.leftwm_versions = theme.leftwm_versions.clone();
                 target_theme.dependencies = theme.dependencies.clone();
+                target_theme.directory = theme.directory.clone();
             }
             // o/w insert a new leaf at the end
             None => {
                 repo.themes.push(theme.clone());
             }
         }
     }
+
+    // Given a leftwm config path and a theme, checks if the theme is installed
+    // by looking for the theme directory and returns the directory path.
+    fn installed_path(config_path: String, name: String) -> Option<PathBuf> {
+        let theme_path = Path::new(&config_path).join(THEMES_DIR).join(name);
+        if theme_path.exists() {
+            Some(theme_path)
+        } else {
+            None
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_installed_path() {
```

Comment:
Thank you for adding a test; I've been meaning to add more for `leftwm-theme` but haven't yet. This is awesome and very "Rust-y".

Reply:
I though that there are no tests because the project may be new and may change a lot for now.
I can try adding more tests and practice more rust :)